backend/reg.ml

```diff
@@ -299,6 +299,15 @@ let set_of_array v =
            if i >= n then Set.empty else Set.add v.(i) (add_all(i+1))
          in add_all 0
 
+let set_has_collisions s =
+  let phys_regs = Hashtbl.create (Set.cardinal s) in
```

Comment:
(I don't think that matters that much,
but on pathological cases the cardinal
of `s` could be very large, while the
cardinal of `phys_regs` is bounded by
the total number of hardware registers.)

Reply:
True, except we can't refer to `Proc.num_available_registers` here without breaking a dependency cycle.
Changed to min with 32 instead